extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -1,35 +1,155 @@
 import * as Preact from '#core/dom/jsx';
 import {Layout_Enum} from '#core/dom/layout';
 
+import {user} from '#utils/log';
+import {Services} from '#service';
+import {LocalizedStringId_Enum} from '#service/localization/strings';
+
+import {formatI18nNumber, loadFonts} from './amp-story-shopping';
+
+import {
+  ShoppingConfigDataDef,
+  StateProperty,
+} from '../../amp-story/1.0/amp-story-store-service';
+
+/** @const {!Array<!Object>} fontFaces */
+const FONTS_TO_LOAD = [
+  {
+    family: 'Poppins',
+    weight: '500',
+    src: "url(https://fonts.gstatic.com/s/poppins/v15/pxiByp8kv8JHgFVrLGT9Z1xlFd2JQEk.woff2) format('woff2')",
+  },
+  {
+    family: 'Poppins',
+    weight: '600',
+    src: "url(https://fonts.gstatic.com/s/poppins/v15/pxiByp8kv8JHgFVrLEj6Z1xlFd2JQEk.woff2) format('woff2')",
+  },
+];
+
+const TAG = 'amp-story-shopping-attachment';
+
 export class AmpStoryShoppingAttachment extends AMP.BaseElement {
   /** @param {!AmpElement} element */
   constructor(element) {
     super(element);
 
     /** @private {?Element} */
     this.attachmentEl_ = null;
+
+    /** @private {!Element} */
+    this.pageEl_ = this.element.closest('amp-story-page');
+
+    /** @private {!Array<!Element>} */
+    this.shoppingTags_ = Array.from(
+      this.pageEl_.querySelectorAll('amp-story-shopping-tag')
+    );
+
+    /** @private @const {!Element} */
+    this.plpContainer_ = <div></div>;
+
+    /** @private {?../../amp-story/1.0/amp-story-store-service.AmpStoryStoreService} */
+    this.storeService_ = null;
+
+    /** @private {?../../../src/service/localization.LocalizationService} */
+    this.localizationService_ = null;
   }
 
   /** @override */
   buildCallback() {
+    loadFonts(this.win, FONTS_TO_LOAD);
     this.attachmentEl_ = (
       <amp-story-page-attachment
         layout="nodisplay"
         theme={this.element.getAttribute('theme')}
       ></amp-story-page-attachment>
     );
     this.element.appendChild(this.attachmentEl_);
+    this.attachmentEl_.appendChild(this.plpContainer_);
+
+    return Promise.all([
+      Services.storyStoreServiceForOrNull(this.win),
+      Services.localizationServiceForOrNull(this.element),
+    ]).then(([storeService, localizationService]) => {
+      this.storeService_ = storeService;
+      this.localizationService_ = localizationService;
+    });
+  }
+
+  /** @override */
+  layoutCallback() {
+    this.storeService_.subscribe(
+      StateProperty.PAGE_ATTACHMENT_STATE,
+      (isOpen) => this.onPageAttachmentStateUpdate_(isOpen)
+    );
+  }
+
+  /**
+   * On attachment state update, check if on active page and populate plp.
+   * @param {boolean} isOpen
+   * @private
+   */
+  onPageAttachmentStateUpdate_(isOpen) {
+    const isOnActivePage =
+      this.pageEl_.id === this.storeService_.get(StateProperty.CURRENT_PAGE_ID);
+    if (isOpen && isOnActivePage) {
+      this.populatePlp_();
+    }
+  }
+
+  /** @private */
+  populatePlp_() {
```

Comment:
Nit: please add a method description, preferably one that includes the words "PLP (Product Listing Page)" so that someone could figure out what "PLP" means within this file